src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -953,29 +969,41 @@ private boolean checkForSaveAndContinue() {
     return willContinue;
   }
 
-  private boolean checkForSaveAndExecute() {
-    boolean willContinue = false;
-    if (checkIfNeedsSaving()) {
+  private String commitConfigToFileAndGetFilepath() {
+    String filename = null;
+    ConfigComparisonResult comparisonResult = compareConfigs();
+    if (comparisonResult != ConfigComparisonResult.SAME) {
       ButtonType saveButton = new ButtonType("Save", ButtonBar.ButtonData.YES);
+      ButtonType dontSaveButton = new ButtonType("Proceed without Saving", ButtonBar.ButtonData.NO);
+      ButtonType cancelButton = new ButtonType("Cancel", ButtonBar.ButtonData.NO);
+      ButtonType noButton = comparisonResult
+          == ConfigComparisonResult.DIFFERENT_BUT_VERSION_IS_TEST ? dontSaveButton : cancelButton;
       Alert alert =
           new Alert(
               AlertType.WARNING,
               "You must either save your changes before continuing or load a new contest config!",
               saveButton,
-              ButtonType.CANCEL);
+              noButton);
       alert.setHeaderText(null);
       Optional<ButtonType> result = alert.showAndWait();
       if (result.isPresent() && result.get() == saveButton) {
         File fileToSave = getSaveFile();
         if (fileToSave != null) {
           saveFile(fileToSave);
-          willContinue = true;
+          filename = fileToSave.getAbsolutePath();
         }
+      } else if (result.get() == dontSaveButton) {
+        File tempFile = new File(selectedFile.getAbsolutePath() + "_temp");
+        filename = tempFile.getAbsolutePath();
+        saveFile(tempFile);
+        tempFile.deleteOnExit();
```

Comment:
Curious why we wait until the program is exited to delete the config?

Reply:
We could also do it when tabulation completes, or when another file is selected, but that would require following this around and adding some complexity since Java doesn't have a destructor. `deleteOnExit` was the cleanest one-line solution.

Reply:
Gotcha, thanks for clarifying. Generally speaking, this SGTM, but I did find a corner case accidentally that's a little annoying for development: if you terminate the program early via the CLI (or in IntelliJ by clicking the stop button), it won't actually clean up after itself and it'll just leave the temp files there indefinitely.

Could you investigate deleting right after tabulation a little bit? If it looks like it'll take more than 10-20 minutes of work, or increase the code complexity significantly / in an ugly fashion, I'm fine leaving it this way and just trying to remember not to stop the tabulator in this way.